src/Castle.Facilities.AspNetCore.Tests/AspNetCoreFacilityTestCase.cs

```diff
@@ -99,7 +99,13 @@ private static ServiceCollection BuildServiceCollection()
 		private void BuildWindsorContainer(ServiceCollection serviceCollection)
 		{
 			container = new WindsorContainer();
-			serviceCollection.AddCastleWindsor(container);
+			serviceCollection.AddCastleWindsor(container,
+				options =>
+				{
+					options.AddHttpContextAccessor();
+					options.AddContainerResolvedScoped<IFakeFrameworkComponent>();
+					options.AddContainerResolvedScoped<IFakeFrameworkComponentEnumerable>();
+				});
```

Comment:
Why can't this be done on the original service collection? Those API's are well documented. All the third party extensions would eventually need to integrate with these API's. What problem this trying to solve?

Reply:
I am just adding into the facity to add extra options for the castle registation so everythng to the service collection is nicely wrapped.
I think it is very powerful and ver neet. Only the facility needst to know how to work with the service collection :-)

Reply:
This is dangerous and could have unintended consequences. Wrapping the IHttpContextAccessor could mean we are taking a dependency on a type which has been flagged for deprecation. Please see: https://github.com/aspnet/Announcements/issues/190